src/bgen/AttributeManager.cs

```diff
@@ -569,23 +563,39 @@ public static bool HasAttribute (ICustomAttributeProvider provider, string type_
 		if (rv.Length == 1)
 			return rv [0];
 
-		string name = (provider as MemberInfo)?.Name;
-		if (provider is ParameterInfo) {
-			var pi = (ParameterInfo) provider;
-			name = $"the method {pi.Member.DeclaringType.FullName}.{pi.Member.Name}'s parameter #{pi.Position} ({pi.Name})";
-		} else if (provider is Type type) {
-			name = $"the type {type.FullName}";
-		} else if (provider is MemberInfo) {
-			var mi = (MemberInfo) provider;
-			name = $"the member {mi.DeclaringType.FullName}.{mi.Name}";
-		} else if (provider is Assembly) {
-			name = $"the assembly {((Assembly) provider).FullName}";
-		} else if (provider is Module) {
-			name = $"the module {((Module) provider).FullyQualifiedName}";
-		} else {
-			name = $"the member {provider.ToString ()}";
+		int code;
+		object [] args;
+		// each type of provider has its own error. This is because each exception has its own message that 
+		// must be correctly translated.
+		switch (provider) {
+		case ParameterInfo pi:
+			code = 1083;
+			args = new object [] {
+				rv.Length, typeof (T).FullName, $"{pi.Member.DeclaringType.FullName}.{pi.Member.Name}", pi.Position, pi.Name
+			};
+			break;
+		case Type type:
+			code = 1084;
+			args = new object [] { rv.Length, typeof (T).FullName, type.FullName };
+			break;
+		case MemberInfo mi:
+			code = 1059;
+			args = new object [] { rv.Length, typeof (T).FullName, $"{mi.DeclaringType?.FullName}.{mi.Name}" };
+			break;
+		case Assembly ass:
+			code = 1085;
+			args = new object [] { rv.Length, typeof (T).FullName, $"{ass.FullName}" };
+			break;
```

Comment:
```suggestion
		case Assembly assm:
			code = 1085;
			args = new object [] { rv.Length, typeof (T).FullName, $"{assm.FullName}" };
			break;
```
I bet policheck complains...